### ISSUE TITLE
Where is file yy_parser.hpp?

### ISSUE THREAD

**Issue by Lakasabasz** [avast-tl/yaramod#6]
When I try to compile get error about missing #include "yaramod/yy/yy_parser.hpp".

**Comment by metthal** [comment]
`yy_parser.hpp` is automatically generated by `bison`. Make sure that you have at least bison version 3 and that running cmake does not produce any errors regarding `flex` and `bison`.

**Comment by Lakasabasz** [comment]
I was rebuilded project file, and I still don't have that file. Can I generate it manually?

**Comment by metthal** [comment]
Well, you can generate it manually by running `bison` with `parser.yy` file but it would be much better to find out why it isn't working for you. Could you please provide information about what OS you use, compiler and also attach build log from clean build?

**Comment by Lakasabasz** [comment]
I was trying to build it because it is one of targets created by cmake for retdec program. When I execute command: `> bison parser.y` I got error about version of bison (it was to low). I downloaded newer version and I try build it one more time.

I am working on Windows 7, with compiler gcc version 7.1. Build log I'll send later

**Comment by Lakasabasz** [comment]
[build log.txt](https://github.com/avast-tl/yaramod/files/2217208/build.log.txt)

**Comment by s3rvac** [comment]
From the attached log file, I can see that there is an error when building `yaracpp`. Could you please provide us the content of the following two log files?
```
G:/sync/retdec-master/external/src/yaracpp-project-stamp/yaracpp-project-build-out.log
G:/sync/retdec-master/external/src/yaracpp-project-stamp/yaracpp-project-build-err.log
```

**Comment by Lakasabasz** [comment]
[yaracpp-project-build-err.log](https://github.com/avast-tl/yaramod/files/2219133/yaracpp-project-build-err.log)
[yaracpp-project-build-out.log](https://github.com/avast-tl/yaramod/files/2219134/yaracpp-project-build-out.log)

**Comment by metthal** [comment]
Based on the files you have provided (`yaramod-project-build-*.log`), it seems like you have outdated version of `flex` installed. Could you please try running `flex --version` in your command line? It should be at least 2.6.

The problems with `yaracpp` build should be reported in [its own repository](https://github.com/avast-tl/yaracpp).

**Comment by Lakasabasz** [comment]
Yes, you're right. Version of my flex is 2.5.4. I'm sorry for yaracpp, but generally have trouble with retdec compilation and here i want to fix errors with yaramod. Now i update flex and try to compile that target. If it will be fixed that issue can be closed.

**Comment by Lakasabasz** [comment]
Unfortunetly, it's not working:
[build log.txt](https://github.com/avast-tl/yaramod/files/2219332/build.log.txt)
[yaramod-project-build-out.log](https://github.com/avast-tl/yaramod/files/2219335/yaramod-project-build-out.log)
[yaramod-project-build-err.log](https://github.com/avast-tl/yaramod/files/2219336/yaramod-project-build-err.log)

**Comment by s3rvac** [comment]
BTW, I have just noticed this:

> I am working on Windows 7, with compiler gcc version 7.1. Build log I'll send later

On Windows, the only compiler that RetDec (and probably also yaramod/yaracpp) officially supports is Microsoft Visual Studio >= 2015. We have no idea whether you can build and run RetDec with GCC on Windows as we only use Visual Studio on Windows.

**Comment by s3rvac** [comment]
No offense taken :). Actually, almost all of us develop on Linux, where we use GCC and Clang.

A few years ago (before RetDec was open-sourced), it was actually possible to build RetDec with GCC on Windows. However, at one point (I think it started with GCC 5), the builds started crashing due to unknown reasons, and we were unable to find the underlying cause. Thus, we removed GCC support from the build system on Windows, and started using only Visual Studio. Frankly speaking, Visual Studio is to Windows what GCC is to Linux and what Clang is to macOS. In the open-sourced version of RetDec, Windows users are thus expected to use Visual Studio. This also simplified the build system as previously, we had to support three variants: Linux, Windows with Visual Studio, and Windows with GCC.

Feel free to open [an issue](https://github.com/avast-tl/retdec/issues) requesting GCC support on Windows. However, I cannot promise that we will be able to add it. Nevertheless, when someone implements this and submits [a PR](https://github.com/avast-tl/retdec/pulls), we will definitely do our best to incorporate that PR into RetDec.
